src/catimg.c

```diff
@@ -67,14 +71,39 @@ int main(int argc, char *argv[])
     int c;
     opterr = 0;
 
-    uint32_t cols = 0, precision = 0;
+    uint32_t cols = 0, rows = 0, precision = 0;
+    uint32_t max_cols = 0, max_rows = 0;
     uint8_t convert = 0;
     uint8_t true_color = 1;
-    while ((c = getopt (argc, argv, "w:l:r:hct")) != -1)
+    uint8_t adjust_to_height = 0, adjust_to_width = 0;
+    float sc = 0, sr = 0;
```

Comment:
can we also use full names for these variables?

Reply:
that can be done